src/main/java/squid/tasks/Tasks.java

```diff
@@ -0,0 +1,197 @@
+        }
+        return res.toString();
+    }
+
+    /**
+     * Prints the tasks contained in the given input.
+     * @param arr The data structure representing tasks.
+     * @return the String representation of current tasks.
+     */
+    public static String list(List<Task> arr) {
+        StringBuilder res = new StringBuilder();
+        for (int i = 0; i < arr.size(); i++) {
+            Task currTask = arr.get(i);
+            res.append(String.format("%d: %s%n", i + 1, currTask));
+        }
+        return res.toString();
+    }
+
+    /**
+     * Finds tasks matching the input.
+     * @param regex the input to match similar tasks with.
+     */
+    public static String find(String regex) {
+        List<Task> filtered = arr.stream().filter(x -> x.getTaskName().matches(".*" + regex + ".*")).toList();
+        return list(filtered);
+    }
+
+    /**
+     * Convert a String from the hard disk storage file into a Task object.
+     * @param s The string to be parsed.
+     * @return The parsed Task.
+     * @throws ParseFailException If the string is unable to be parsed.
+     * @throws SquidDateException If the DateTime object is unable to be parsed.
+     */
+    public static Task parseTask(String s) throws ParseFailException, SquidDateException {
+        String[] params = s.split(Regex.TASK_SPLIT);
+        Task task;
+        switch (params.length) {
+        case (5): // event
+            task = new Event(params[2], new DateTime(params[3]), new DateTime(params[4]));
+            task.setCompleted(Objects.equals(params[1], "X"));
+            return task;
+        case (4): // deadline
+            task = new Deadline(params[2], new DateTime(params[3]));
+            task.setCompleted(Objects.equals(params[1], "X"));
+            return task;
+        case (3): //todo
+            task = new Todo(params[2]);
+            task.setCompleted(Objects.equals(params[1], "X"));
+            return task;
+        default:
+            throw new ParseFailException(String.format(Exceptions.PARSE_FAIL, s));
+        }
+    }
+
+    /**
+     * Saves the current list of tasks onto the hard disk.
+     */
+    public static void save() {
+        StringBuilder stringBuilder = new StringBuilder();
+        for (int i = 0; i < arr.size(); i++) {
+            String s = arr.get(i).parseStr();
+            stringBuilder.append((s));
+        }
+        assert(SquidFile.doesFileExist(Filepath.FULL_PATH));
+        try {
+            SquidFile.writeToFile(stringBuilder.toString(), false);
+        } catch (IOException ignored) {
+            ignored = ignored;
+        }
+    }
+
+    /**
+     * Reads current tasks from hard disk storage.
+     * @throws ParseFailException If a string is unable to be parsed.
+     * @throws DuplicateTaskNameException If there are duplicate task names in the storage file.
+     * @throws SquidDateException If a DateTime object is unable to be parsed.
+     */
```

Comment:
I like the clarity here for the javadocs, and it's nice that it's properly aligned too!